mesh/reward.go

```diff
@@ -0,0 +1,100 @@
+package mesh
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/state"
+	"math/big"
+)
+
+type RewardParams struct {
+	SimpleTxCost   *big.Int
+	BaseReward     *big.Int
+	PenaltyPercent *big.Int
+	TxQuota        uint32
+}
+
+//type Transactions []*state.Transaction
+
+func MergeDoubles(transactions []*state.Transaction) []*state.Transaction {
+	transactionSet := make(map[common.Hash]struct{})
+	merged := make([]*state.Transaction, 0, len(transactions))
+	for _, trns := range transactions {
+		if _, ok := transactionSet[trns.Hash()]; !ok {
+			transactionSet[trns.Hash()] = struct{}{}
+			merged = append(merged, trns)
+		} else {
+			log.Info("double trans merged %v", trns)
+		}
+	}
+	return merged
+}
+
+func (m *Mesh) CalculateLayerReward(id LayerID, params RewardParams) *big.Int {
```

Comment:
CalculateLayerBaseReward

Reply:
why calculate if it just returns LayerReward ?
also why dose this have a mesh receiver ? it doesn't use mesh at all